Src/DependencyCollector/Shared.Tests/Implementation/ProfilerSqlProcessingTest.cs

```diff
@@ -143,6 +143,22 @@ public void OnEndSendsCorrectTelemetry_1ArgumentOverride()
         }
 
         [TestMethod]
```

Comment:
you need a test to validate that `End` for dependency telemetry will in fact be called